test-framework/mongodb/src/main/java/io/quarkus/test/mongodb/MongoTestResource.java

```diff
@@ -1,117 +1,53 @@
 package io.quarkus.test.mongodb;
 
-import static de.flapdoodle.embed.process.config.process.ProcessOutput.builder;
-
-import java.io.IOException;
-import java.util.Collections;
-import java.util.Map;
-import java.util.Optional;
-
-import org.jboss.logging.Logger;
-
-import de.flapdoodle.embed.mongo.Command;
-import de.flapdoodle.embed.mongo.MongodExecutable;
-import de.flapdoodle.embed.mongo.MongodStarter;
-import de.flapdoodle.embed.mongo.config.Defaults;
-import de.flapdoodle.embed.mongo.config.MongoCmdOptions;
-import de.flapdoodle.embed.mongo.config.MongodConfig;
+import de.flapdoodle.embed.mongo.commands.MongodArguments;
 import de.flapdoodle.embed.mongo.config.Net;
 import de.flapdoodle.embed.mongo.distribution.IFeatureAwareVersion;
-import de.flapdoodle.embed.mongo.distribution.Version;
-import de.flapdoodle.embed.process.config.RuntimeConfig;
-import de.flapdoodle.embed.process.io.Processors;
-import de.flapdoodle.embed.process.runtime.Network;
+import de.flapdoodle.embed.mongo.transitions.Mongod;
+import de.flapdoodle.embed.mongo.transitions.RunningMongodProcess;
+import de.flapdoodle.reverse.TransitionWalker;
+import de.flapdoodle.reverse.transitions.Start;
 import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import org.jboss.logging.Logger;
 
-public class MongoTestResource implements QuarkusTestResourceLifecycleManager {
-    public static final String PORT = "port";
-    public static final String VERSION = "version";
-    private static final int DEFAULT_PORT = 27017;
-    private static MongodExecutable MONGO;
+import java.util.Collections;
+import java.util.Map;
 
+public class MongoTestResource implements QuarkusTestResourceLifecycleManager {
     private static final Logger LOGGER = Logger.getLogger(MongoTestResource.class);
     private Integer port;
     private IFeatureAwareVersion version;
 
+    private TransitionWalker.ReachedState<RunningMongodProcess> started;
```

Comment:
I'd rather named it `mongo` than started which look like a boolean

Reply:
@loicmathieu ah, ok.. will change this.